### PR TITLE
gdbstub: doc: Update documentation with example

### LINKED ISSUE #70966: doc: Documentation issue in 'services/debugging/gdbstub'
**Describe the bug**

Is this feature, gdbstub, supported on Zephyr 3.6.99?
Since I'm not able to find a directory under "samples/subsys/debug/gdbstub".
Thank you,

Kevin

<< Please describe the issue here >>
<< You may also want to update the automatically generated issue title above. >>

**Environment**

* Page: `services/debugging/gdbstub`
* Version: main
* SHA-1: fb19d532edb177dda56fb0f37c89f361004f41c0

### PR DESCRIPTION
**Pull request by golowanow**
Update the Zephyr `gdbstub` documentation with its 'how to use' example to align with former sample application replaced by the test.

Fixes #70966

### CODE REVIEW

**Comment by golowanow** [comment]
the updated page:
https://builds.zephyrproject.io/zephyr/pr/71063/docs/services/debugging/gdbstub.html

**Review by nashif — DISMISSED** [review]

**Review by dcpleung — DISMISSED** [review]

**Review by kartben — CHANGES_REQUESTED** [review]

File: doc/services/debugging/gdbstub.rst [hunk 1]
```diff
@@ -87,20 +87,39 @@ Using Serial Backend
...
 
-This is an example to demonstrate how GDB stub works.
-You can also refer to ``tests/subsys/debug/gdbstub``
-for its implementation as a Twister test.
+There is a test application :zephyr_file:`tests/subsys/debug/gdbstub` with one of its
+test cases ``debug.gdbstub.breakpoints`` demonstrating how the Zephyr GDB stub can be used.
+The test also has a case to connect to the QEMU's GDB stub implementation (at a custom
+port ``tcp:1235``) as a reference to validate the test script itself.
+
+Run the test with the following command at your :envvar:`ZEPHYR_BASE` directory:
```
[the remark this answers is not in the document]
Comment (golowanow) [reply doc/services/debugging/gdbstub.rst]: done

File: doc/services/debugging/gdbstub.rst [hunk 2]
```diff
@@ -125,49 +144,52 @@ for its implementation as a Twister test.
 
          (gdb) target remote localhost:5678
 
-      Note that QEMU is setup to redirect the serial used for GDB stub in
-      the Zephyr image to a networking port. Hence the connection to
-      localhost, port 5678.
+      The QEMU is setup earlier to redirect its serial console port for
+      the GDB stub in the Zephyr image to a network port.
+      Hence the GDB remote connection is to ``localhost``, port ``5678``.
```
[the remark this answers is not in the document]
Comment (golowanow) [reply doc/services/debugging/gdbstub.rst]: ok, changed.

File: doc/services/debugging/gdbstub.rst [hunk 3]
```diff
@@ -287,13 +290,15 @@ for its implementation as a Twister test.
...
+         29              printk("ret=%d\n", ret);
          (gdb) p ret
-         $4 = 0x1e
+         $4 = 30
 
    #. If ``continue`` is issued here, code execution will continue indefinitely
       as there are no breakpoints to further stop execution. Breaking execution
-      in GDB via Ctrl-C does not currently work as the GDB stub does not
-      support this functionality (yet).
+      in GDB via ``Ctrl-C`` does not currently work as the GDB stub does not
```
[the remark this answers is not in the document]
Comment (golowanow) [reply doc/services/debugging/gdbstub.rst]: changed

File: doc/services/debugging/gdbstub.rst [hunk 4]
```diff
@@ -287,13 +290,15 @@ for its implementation as a Twister test.
...
+      support this functionality (yet). Switch to the first console with QEMU
+      running the Zephyr image and stop it manually with ``Ctrl+a x``.
```
[the remark this answers is not in the document]
Comment (golowanow) [reply doc/services/debugging/gdbstub.rst]: changed

File: doc/services/debugging/gdbstub.rst [hunk 5]
```diff
@@ -287,13 +290,15 @@ for its implementation as a Twister test.
...
+      When the same test is executed by Twister it takes care of stopping the QEMU.
```
[the remark this answers is not in the document]
Comment (golowanow) [reply doc/services/debugging/gdbstub.rst]: changed

**Review by dcpleung — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Comment by nashif** [comment]
@kartben  please revisit

File: doc/services/debugging/gdbstub.rst [hunk 6]
```diff
@@ -111,63 +130,63 @@ for its implementation as a Twister test.
...
 
       Response from GDB:
 
       .. code-block:: text
 
          Reading symbols from <build directory>/zephyr/zephyr.elf...
 
-   #. Tell GDB to connect to the server:
+   #. Tell GDB to connect to the Zephyr gdbstub serial backend which is exposed
+      earlier as a server through the TCP port ``-serial`` redirection at QEMU.
```
Comment (kartben) [inline doc/services/debugging/gdbstub.rst:142]: Not sure this works better than the wording I had proposed (ex. I think "at QEMU" isn't proper English)

**Review by kartben — APPROVED** [review]

**Pull request merged**